Flat rides

pollend opened the pull request:
Here is the flatride extension I've been working on. I'm not quite done yet but this is what I have finished so far. I still need to add some comment and I'm still working on a couple adjustments.  In terms of importing the ride into Parkitect, I would look at the sample proxy object I wrote. 


https://github.com/pollend/Parkitect_Mod_Bootstrap/blob/master/Model/Proxy/FlatRide/CustomFlatRide.cs

pollend committed:
started on asset type

pollend committed:
added attraction type

pollend committed:
added sitting refrence guest and added some basic comments

Sebioff reviewed (commented):
- BenchRenderer -> SeatRenderer
- AssetType.Flatride -> AssetType.FlatRide
- AssetEditorWindow.drawFlatRideSection -> AssetEditorWindow.DrawAssetFlatRideDetailSection

- AssetEditorWindow.GetRatingCategory returns incorrect categories that don't match the ones in the game

pollend committed:
BenchRenderer -> SeatRenderer
AssetType.Flatride -> AssetType.FlatRide
AssetEditorWindow.drawFlatRideSection -> AssetEditorWindow.DrawAssetFlatRideDetailSection
updated rating category.

pollend committed:
removed category and scaled display values for Excitment Intensity and Nausea by a factor of 100

pollend committed:
Added bounded box collider support

Sebioff commented:
Merged to feature/flat_rides
